src/Expander.hs

```diff
@@ -1058,26 +1095,23 @@ expandOneForm prob stx
                   else for (zip args' foundArgs) (uncurry schedule)
               linkExpr dest (CoreCtor ctor argDests)
               saveExprType dest t
-            PatternDest _exprTy patTy dest -> do
-              Stx _ loc (_cname, patVars) <- mustBeCons stx
+            PatternDest exprTy patTy dest -> do
```

Comment:
I don't think this `exprTy` is used anywhere, is it? Here, we just forward it along in the `PatternDest` of the nested patterns.

Reply:
It was indeed unused, I removed it in a subsequent PR.

Reply:
Nice!